### PR TITLE
WFLY-18404 HotRod-based session manager creates too many threads for handling concurrent expiration events

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-18404

Requires (not really, but wanted to avoid merge conflicts) https://github.com/wildfly/wildfly/pull/17139

### CODE REVIEW

File: clustering/web/hotrod/src/main/java/org/wildfly/clustering/web/hotrod/session/HotRodSessionMetaDataFactory.java [hunk 1]
```diff
@@ -48,28 +48,23 @@
 import org.wildfly.clustering.web.session.ImmutableSessionMetaData;
 
 /**
+ * Factory for creating {@link SessionMetaData} backed by a pair of {@link RemoteCache} entries.
  * @author Paul Ferraro
+ * @param <C> the local context type
```
Comment (rhusar) [inline clustering/web/hotrod/src/main/java/org/wildfly/clustering/web/hotrod/session/HotRodSessionMetaDataFactory.java:53]: Another s/C/L here - but we don't need to fix this now.

**Review by rhusar — APPROVED** [review]
LGTM

**Comment by pferraro** [comment]
Rebased against updated WFLY-18389 branch.

File: clustering/web/hotrod/src/main/java/org/wildfly/clustering/web/hotrod/session/HotRodSessionMetaDataFactory.java (hunk 1 above)
Comment (pferraro) [reply clustering/web/hotrod/src/main/java/org/wildfly/clustering/web/hotrod/session/HotRodSessionMetaDataFactory.java:53]: Fixed

**Pull request merged**